src/File.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+declare(strict_types=1);
+
+namespace IterTools;
+
+class File
+{
+    /**
+     * Reads file resource line by line.
+     *
+     * @param resource $fileResource
+     * @param int<0, max>|null $length
+     *
+     * @return \Generator<string>
+     *
+     * @see fgets()
+     */
+    public static function readByLine($fileResource, ?int $length = null): \Generator
```

Comment:
What do you think about getting rid of the `$length` parameter to simplify this. I'm pretty sure 99% of users will not use it.

Reply:
Agree, done.